core/arch/arm/include/arm64.h

```diff
@@ -366,6 +366,17 @@ static inline __noprof void tlbi_vale1is(uint64_t va)
 	asm volatile ("tlbi	vale1is, %0" : : "r" (va));
 }
 
+static inline __noprof uint32_t read_isar5(void)
+{
+	uint32_t isar_reg;
+#if defined(__clang__)
+	asm volatile("mrs %w0, ID_ISAR5_EL1" : "=r"(isar_reg));
```

Comment:
Added this hack for clang to suppress the error below. Looks like it's a false positive, as `ID_ISAR5_EL1` is 32bit register:
```
core/arch/arm/include/arm64.h:373:45: error: value size does not match register size specified by the constraint and modifier [-Werror,-Wasm-operand-widths]
        asm volatile("mrs %0, ID_ISAR5_EL1" : "=r"(isar_reg));
                                                   ^
core/arch/arm/include/arm64.h:373:20: note: use constraint modifier "w"
        asm volatile("mrs %0, ID_ISAR5_EL1" : "=r"(isar_reg));
                          ^~
                          %w0
1 error generated.
```

and gcc doesn't support that syntax (which works for clang), reporting:
```
{standard input}: Assembler messages:
{standard input}:1822: Error: operand mismatch -- `mrs w0,ID_ISAR5_EL1'
{standard input}:1822: Info:    did you mean this?
{standard input}:1822: Info:    	mrs x0, id_isar5_el1

```

Let me know if you have a better idea how to address that